nav2_map_server/src/vo_server/vector_object_server.cpp

```diff
@@ -0,0 +1,582 @@
+// Copyright (c) 2023 Samsung R&D Institute Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_map_server/vector_object_server.hpp"
+
+#include <uuid/uuid.h>
+#include <cmath>
+#include <functional>
+#include <limits>
+#include <stdexcept>
+
+#include "tf2_ros/create_timer_ros.h"
+
+#include "nav2_util/occ_grid_utils.hpp"
+
+using namespace std::placeholders;
+
+namespace nav2_map_server
+{
+
+VectorObjectServer::VectorObjectServer(const rclcpp::NodeOptions & options)
+: nav2_util::LifecycleNode("vector_object_server", "", options), process_map_(false)
+{}
+
+nav2_util::CallbackReturn
+VectorObjectServer::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Configuring");
+
+  // Transform buffer and listener initialization
+  tf_buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
+  auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+    this->get_node_base_interface(),
+    this->get_node_timers_interface());
+  tf_buffer_->setCreateTimerInterface(timer_interface);
+  tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*tf_buffer_);
+
+  // Obtaining ROS parameters
+  if (!getROSParameters()) {
+    return nav2_util::CallbackReturn::FAILURE;
+  }
+
+  map_pub_ = create_publisher<nav_msgs::msg::OccupancyGrid>(
+    map_topic_,
+    rclcpp::QoS(rclcpp::KeepLast(1)).transient_local().reliable());
+
+  // Make name prefix for services
+  const std::string service_prefix = get_name() + std::string("/");
+
+  add_shapes_service_ = create_service<nav2_msgs::srv::AddShapes>(
+    service_prefix + std::string("add_shapes"),
```

Comment:
`~/add_shapes` does the same thing without needing `  const std::string service_prefix = get_name() + std::string("/");`

Reply:
Done in https://github.com/ros-planning/navigation2/pull/3930/commits/d6a831d90d1440436538a61c771f017bddb59422